src/memberBundle.ts

```diff
@@ -0,0 +1,175 @@
+import { getModelForClass } from '@typegoose/typegoose';
+import { addDays } from 'date-fns';
+import {
+  issueMemberCertificate,
+  retrieveVeraDnssecChain,
+  selfIssueOrganisationCertificate,
+  serialiseMemberIdBundle,
+} from '@relaycorp/veraid';
+import type { BaseLogger } from 'pino';
+
+import type { ServiceOptions } from './serviceTypes.js';
+import { MemberPublicKeyModelSchema } from './models/MemberPublicKey.model.js';
+import { MemberModelSchema } from './models/Member.model.js';
+import { Kms } from './utilities/kms/Kms.js';
+import { OrgModelSchema } from './models/Org.model.js';
+import { derDeserialisePublicKey } from './utilities/webcrypto.js';
+import type { Result } from './utilities/result.js';
+
+const CERTIFICATE_EXPIRY_DAYS = 90;
+interface CreateBundleInput {
+  orgPrivateKeyRefBuffer: Buffer;
+  orgPublicKeyBuffer: Buffer;
+  memberPublicKey: Buffer;
+  orgName: string;
+  memberName?: string;
+  memberPublicKeyId: string;
+}
+
+async function generateBundle(
+  {
+    orgPrivateKeyRefBuffer,
+    orgPublicKeyBuffer,
+    memberPublicKey,
+    orgName,
+    memberName,
+    memberPublicKeyId,
+  }: CreateBundleInput,
+  logger: BaseLogger,
+): Promise<ArrayBuffer | undefined> {
+  const kms = await Kms.init();
+  const orgPrivateKey = await kms.retrievePrivateKeyByRef(orgPrivateKeyRefBuffer);
+  const orgPublicKey = await derDeserialisePublicKey(orgPublicKeyBuffer);
+  const orgKeyPair: CryptoKeyPair = {
+    privateKey: orgPrivateKey,
+    publicKey: orgPublicKey,
+  };
+
+  const memberCryptoPublicKey = await derDeserialisePublicKey(memberPublicKey);
+
+  const expiryDate = addDays(new Date(), CERTIFICATE_EXPIRY_DAYS);
+  const orgCertificate = await selfIssueOrganisationCertificate(orgName, orgKeyPair, expiryDate);
+
+  const memberCertificate = await issueMemberCertificate(
+    memberName,
+    memberCryptoPublicKey,
+    orgCertificate,
+    orgPrivateKey,
+    expiryDate,
+  );
+
+  let dnssecChain;
+  try {
+    dnssecChain = await retrieveVeraDnssecChain(orgName);
```

Comment:
Sorry, I missed this in the review I did last night:

We should be retrieving the DNSSEC chain before we try to do anything with the KMS: The DNSSEC chain retrieval is expensive CPU- and bandwidth-wise, as well as likely to fail due to the **many** HTTP requests it makes behind the scenes.

So we shouldn't try to use the KMS until we actually have a DNSSEC chain: KMS operations are expensive -- financially and computationally.

Please also write a test for this so we won't introduce a regression in the future.

TL;DR: Move `retrieveVeraDnssecChain()` to the top of this function.